sentry-android-core/src/main/java/io/sentry/android/core/AppLifecycleIntegration.java

```diff
@@ -44,15 +47,15 @@ public void register(final @NotNull IHub hub, final @NotNull SentryOptions optio
       try {
         Class.forName("androidx.lifecycle.DefaultLifecycleObserver");
         Class.forName("androidx.lifecycle.ProcessLifecycleOwner");
-        watcher =
-            new LifecycleWatcher(
-                hub,
-                this.options.getSessionTrackingIntervalMillis(),
-                this.options.isEnableSessionTracking(),
-                this.options.isEnableAppLifecycleBreadcrumbs());
-        ProcessLifecycleOwner.get().getLifecycle().addObserver(watcher);
-
-        options.getLogger().log(SentryLevel.DEBUG, "AppLifecycleIntegration installed.");
+        if (MainThreadChecker.isMainThread()) {
```

Comment:
`l`: Just out of curiosity, why don't we just always do?
```
final Handler handler = new Handler(Looper.getMainLooper());
handler.post(() -> addObserver(hub));
```

Reply:
if it's already in the main thread, we know that the code will be executed right away.
using a handler, we don't control when this code will be executed (most probably immediately), but let's say that an App. uses `handler/post` heavily, we might be delaying this execution for a bit and things might get out of control, just because of that wdyt?

Reply:
Make sense for me to keep this little extra complexity for such an edge case. Thanks for explaining.